platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -171,137 +230,269 @@ public Schema getSchema(CFMetaData metadata) {
-                boolean[] columnToLoad = new boolean[columnCountToLoad];
-                Arrays.fill(columnToLoad, Boolean.TRUE);
-
-                ISSTableScanner currentScanner = this.ssTableReader.getScanner();
-                Spliterator<UnfilteredRowIterator> splititer = Spliterators.spliteratorUnknownSize(currentScanner,
-                    Spliterator.IMMUTABLE);
-                Stream<UnfilteredRowIterator> partitions = StreamSupport.stream(splititer, false);
-
-                List<IAppendableColumn> arrColumns = this.loadColumns(columnToLoad, partitions, columnCountToLoad);
-                return new Table(arrColumns, this.ssTablePath, null);
+                List<IAppendableColumn> columns = createColumns(
+                        Converters.checkNull(this.actualSchema.getColumnNames()));
+                CassandraSSTableLoader.loadColumns(this.ssTableReader, columns, this.columnDefinitions,
+                        this.tokenRanges, this.localEndpoint);
+                return new Table(columns, this.ssTablePath, null);
             }
         } catch (Exception ex) {
             throw new RuntimeException(ex);
         }
     }
 
-    /** Lazy and non-lazy loader will call this function to load specific column marked by columnToLoad */
-    private List<IAppendableColumn> loadColumns(boolean[] columnToLoad, Stream<UnfilteredRowIterator> partitions, int size) {
-        List<IAppendableColumn> columns = createColumns(columnToLoad, size);
-        // Iterating each item inside the table
-        partitions.forEach(partition -> {
-            while (partition.hasNext()) {
-                Unfiltered unfiltered = partition.next();
-                if (unfiltered instanceof Row) {
-                    Row row = (Row) unfiltered;
-                    int i = 0;
-                    int currentColumn = 0;
-                    Object value;
-                    IAppendableColumn col;
-                    // Extracting the value for each column from a single row
-                    for (ColumnData cd : row) {
-                        // Filter the column to only load the one marked true by columnToLoad[]
-                        if (columnToLoad[i]) {
-                            if (cd.column().isSimple()) {
-                                Cell cell = (Cell) cd;
-                                AbstractType<?> cellType = cell.column().cellValueType();
-                                value = cellType.getSerializer().deserialize(cell.value());
-                            } else {
-                                // Hillview won't process the complex data
-                                throw new RuntimeException("Hillview can't convert complex data");
+    /**
+     * Lazy and non-lazy loader will call this function to load specific column
+     * marked by columnToLoad
+     */
+    private static void loadColumns(SSTableReader ssTableReader, List<IAppendableColumn> columns,
```

Comment:
Since you access columns in the innermost look it may be faster to make it a raw array.

Reply:
But, I need to convert it back to List at the end because of this override:
```
public interface IColumnLoader {
List<? extends IColumn> loadColumns(List<String> names);
}
```
Converting it back will not add significant overhead, right?